server/src/test/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java

```diff
@@ -1639,7 +1641,15 @@ public void testSnapshotWithMissingShardLevelIndexFile() throws Exception {
         try (Stream<Path> files = Files.list(repo.resolve("indices"))) {
             files.forEach(indexPath -> {
                 try {
-                    Files.delete(indexPath.resolve("0").resolve("index-0"));
+                    final String shardGen;
+                    try (Stream<Path> shardFiles = Files.list(indexPath.resolve("0"))) {
+                        shardGen = shardFiles
+                            .filter(file -> file.getFileName().toString().startsWith(BlobStoreRepository.INDEX_FILE_PREFIX))
+                            .findFirst()
+                            .orElseThrow(() -> new AssertionError("Failed to find shard index blob"))
+                            .getFileName().toString().substring(BlobStoreRepository.INDEX_FILE_PREFIX.length());
```

Comment:
Maybe return full absolute path (or relative file name) instead  and pass it to `Files.delete()`?